src/storage/v2/indices.cpp

```diff
@@ -390,15 +390,31 @@ bool LabelPropertyIndex::Entry::operator<(const PropertyValue &rhs) { return val
 bool LabelPropertyIndex::Entry::operator==(const PropertyValue &rhs) { return value == rhs; }
 
 void LabelPropertyIndex::UpdateOnAddLabel(LabelId label, Vertex *vertex, const Transaction &tx) {
-  for (auto &[label_prop, storage] : index_) {
-    if (label_prop.first != label) {
-      continue;
-    }
-    auto prop_value = vertex->properties.GetProperty(label_prop.second);
-    if (!prop_value.IsNull()) {
+  auto vertex_properties = vertex->properties.Properties();
```

Comment:
Deserializing all of the properties might be more expensive than iterating over indices. For example in case of "heap allocated types", such as lists, strings and maps, the deserialization can be costly. We (me and @antoniofilipovic) cannot guess how expensice it is in practice, but it might make sense to measure that somehow. It it turns out to be slow, then maybe we can store the number of properties also in the `PropertyStore`, either as a member variable or at the beginning of the buffer encoded as a variable length integer.